### ISSUE TITLE
AdapterConfig's leave_out not work well in EncoderDecoderModel

### ISSUE THREAD

**Issue by ZeguanXiao** [adapter-hub/adapter-transformers#472]
## Environment info

     
- `adapter-transformers` version: 3.1.0
- Platform: Ubuntu 18.04 (Linux-5.4.0-87-generic-x86_64-with-glibc2.27)
- Python version: Python 3.9.13
- PyTorch version (GPU?): 1.13.1 (GPU)
- Tensorflow version (GPU?): False
- Using GPU in script?: True
- Using distributed or parallel set-up in script?: Yes

## Information

Model I am using (Bert, XLNet ...): EncoderDecoderModel

Language I am using the model on (English, Chinese ...): English

Adapter setup I am using (if any): AdapterConfig

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

**Comment by ZeguanXiao** [comment]
When I check the code, it seems that model.encoder.config.adapter and model.decoder.config.adapter are not referring to the same objects, which is expected to be [here](https://github.com/adapter-hub/adapter-transformers/blob/56be559005089198ac9558cedd0c60ae4e055c9d/src/transformers/adapters/wrappers/configuration.py#L85).

**Comment by ZeguanXiao** [comment]
Also, it seems `EncoderDecoderModelAdaptersMixin.iter_layers` should count decoder layer_id starting with len(self.encoder.layers)).

def iter_layers(self) -> Iterable[Tuple[int, nn.Module]]:
        for i, layer in enumerate(self.encoder.layers):
            yield i, layer
        for i, layer in enumerate(self.decoder.layers, start=len(self.encoder.layers)):
            yield i, layer

**Comment by hSterz** [comment]
Hey @ZeguanXiao, I see why this is unexpected behavior. Unfortunately it is not as easy as changing the `iter_layer` indices. I will look into this.

**Comment by adapter-hub-bert** [comment]
This issue has been automatically marked as stale because it has been without activity for 90 days. This issue will be closed in 14 days unless you comment or remove the stale label.
